Jenkinsfile

```diff
@@ -534,7 +534,15 @@ pipeline {
 
         stage('On a master build') {
           when { branch "master" }
-
+          // Parallel publish edge tag, latest green main builds, to local registry 
+          steps {
+            sh './push-image.sh --edge --registry-prefix=registry.tld'
+          }
+        }
+        
+        stage('On a master build') {
+          when { branch "master" }
+          // Parallel publish edge tag, latest green main builds, to DockerHub
           steps {
             sh './push-image.sh --edge'
           }
```

Comment:
this seems odd. can we have an additional stage inside `On a master build` that will run the tasks in parallel instead? something like this: https://github.com/cyberark/secrets-provider-for-k8s/blob/master/Jenkinsfile#L113

Reply:
@orenbm As I follow existing parallel stage approach in this Jenkisfile I suggest to raise the option for converting it to `tasks` in a follow-up PR that will align all Jenkisfiles if chosen to be our best practice guideline, WDYT?

Reply:
yes but in this case you have 2 things that you want to do in the same scenario - on a master build.
I would at least change the stage names so they are not the same.

Reply:
i still think that we should run these tasks in parallel inside the stage. you can also you the current mechanism to do that by adding an additional `parallel` inside the stage.